yew-macro/src/html_tree/html_tag/mod.rs

```diff
@@ -51,16 +51,19 @@ impl Parse for HtmlTag {
             });
         }
 
-        // Void elements should not have children.
-        // See https://html.spec.whatwg.org/multipage/syntax.html#void-elements
-        match open.tag_name.to_string().as_str() {
-            "area" | "base" | "br" | "col" | "embed" | "hr" | "img" | "input" | "link" | "meta"
-            | "param" | "source" | "track" | "wbr" => {
-                return Err(syn::Error::new_spanned(&open, format!("the tag `<{}>` is a void element and cannot have children (hint: rewrite this as `<{0}/>`)", open.tag_name)));
+        if let TagName::Lit(name) = &open.tag_name {
```

Comment:
I didn't really think about this too much because it's possible to create these elements manually anyway.
The way I see it there are two options:
1. Change `VTag` to check whether it can have children or not and have a "private" method for the macro so the literal case doesn't incur the performance penalty.
2. Generate extra code for dynamic tags that does this check locally (but still at runtime)

The 1st case definitely needs an issue but if you want to go with second case I can just include it here.

Reply:
> Generate extra code for dynamic tags that does this check locally (but still at runtime)

Inline check sounds good to me

> I didn't really think about this too much because it's possible to create these elements manually anyway.

Yeah, this is true. I should have realized this issue earlier 😅